xshop/users/tests/test_models.py

```diff
@@ -162,10 +180,15 @@ def test_created_general_manager_attrs(self):
         self.assertEqual(self.gm.mobile, "+201010092181")
         self.assertTrue(self.gm.is_active)
         self.assertFalse(self.gm.is_staff)
+        self.assertEqual(self.gm.shop, self.shop)
         self.assertFalse(self.gm.is_superuser)
         self.assertIsNone(self.gm.username)
         self.assertEqual(GeneralManager.objects.count(), 1)
 
+    def test_valid_err_if_noshop_with_gm(self):
+        with self.assertRaises(ValidationError):
```

Comment:
make sure the message is the same as the raised error's,
check here for more:
https://docs.djangoproject.com/en/3.1/topics/testing/tools/#assertions